### ISSUE TITLE
Support JDK above 8.x

### ISSUE THREAD

**Issue by seanrohead** [alexandrnikitin/bloom-filter-scala#46]
I'm using JDK 11 (azul 11.0.7) with Scala 2.12.11 and get the exception below when adding an element to the Bloom Filter. The issue appears to be that the value field of class String was changed in JDK 9 from a char[] to a byte[], which causes a ClassCastException on this line:

`  val value = unsafe.getObject(from, valueOffset).asInstanceOf[Array[Char]]`

```
Exception in thread "main" java.lang.ClassCastException: class [B cannot be cast to class [C ([B and [C are in module java.base of loader 'bootstrap')
	at bloomfilter.CanGenerateHashFrom$CanGenerateHashFromString$.generateHash(CanGenerateHashFrom.scala:27)
	at bloomfilter.CanGenerateHashFrom$CanGenerateHashFromString$.generateHash(CanGenerateHashFrom.scala:20)
	at bloomfilter.mutable.BloomFilter.add(BloomFilter.scala:18)
	at Main$.delayedEndpoint$Main$1(Main.scala:12)
	at Main$delayedInit$body.apply(Main.scala:3)
	at scala.Function0.apply$mcV$sp(Function0.scala:39)
	at scala.Function0.apply$mcV$sp$(Function0.scala:39)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:17)
	at scala.App.$anonfun$main$1$adapted(App.scala:80)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.App.main(App.scala:80)
	at scala.App.main$(App.scala:78)
	at Main$.main(Main.scala:3)
	at Main.main(Main.scala)

```

**Comment by alexandrnikitin** [comment]
@seanrohead Thank you for reporting this issue. Hash generating logic is injected via implicits and the type class pattern. You can provide your own implementation. This is what included by default:
https://github.com/alexandrnikitin/bloom-filter-scala/blob/master/bloom-filter/src/main/scala/bloomfilter/CanGenerateHashFrom.scala#L20-L30

As for JDKs other than 8 in general, it's a bit tricky as we need to get rid of all `unsafe` code. I don't a plan (and a need) for that yet.

**Comment by seanrohead** [comment]
@alexandrnikitin Thanks for responding so quickly. I just created a pull request. There were rumors that Unsafe was going to be removed in JDK 9, but they ultimately backed away from that so it's actually quite easy to support both 8 and 9+ (I only tested it on 11, so it's possible that it's broken in some JDK's beyond that).

https://github.com/alexandrnikitin/bloom-filter-scala/pull/47

**Comment by alexandrnikitin** [comment]
@seanrohead That's awesome! Thank you for the contribution!

Shame on me, I stopped following the jdk development (we are stuck with jdk 8). 😊 I finally got myself together and upgraded sbt, also added jdk 10 and 11 to the CI. I'll release the fix tomorrow.

**Comment by alexandrnikitin** [comment]
Closed by #47

**Comment by seanrohead** [comment]
Glad I could help! Thanks for the awesome library 👍

**Comment by alexandrnikitin** [comment]
I finally published `v0.13.0` with your contribution. I'm sorry it took longer. I stumbled over sbt-pgp/sbt-gpg, gpg and travis updates. Thank you for your help!

**Comment by seanrohead** [comment]
Uh oh, I just realized that my fix broke deserialization of old Bloom Filters due to renaming the CanGenerateHashFromString class. Sorry I didn't catch this sooner...

https://github.com/alexandrnikitin/bloom-filter-scala/pull/52
